### PR TITLE
return mouse absolute position

### PR DESCRIPTION
**Pull request by micuat**
dropping mouse-change from package.json and copy them to local lib, in order to change the mouse acquisition to return the absolute coordinate in the window instead of the individual element.

for testing:

```html
<!DOCTYPE html>
<html class="w-100 h-100 mw-100">
  <head>
    <title></title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />
    <meta charset="utf-8">
  </head>
  <body class="w-100 h-100 mw-100 avenir bg-dark-gray near-white">
  	 <div></div>
       <script src='./hydra-synth.js'></script>

    <p id="mouseInfo">
      mouseInfo
    </p>
    <canvas id="myCanvas" width="400px" height="400px"></canvas>
    <script>

window.onload = function () {
  const hydra = new Hydra({canvas: document.getElementById("myCanvas"),
  w: document.body})

  // by default, hydra makes everything global.
  // see options to change parameters
  osc().out()
  update = () => {
        document.querySelector("#mouseInfo").innerHTML = mouse.x + " " + mouse.y;//JSON.stringify(mouse)
      }
    }


       </script>
  </body>
</html>

```

TODO

* should the origin (0, 0) be the left top of the hydra canvas? (p5.js does this, and returns negative when the mouse is left/above the canvas https://editor.p5js.org/micuat/sketches/VkCCISre4 )
* eventually mouse dependency should be dropped from hydra-synth and moved to the editor in my opinion

### CODE REVIEW

**Comment by ojack** [comment]
OOoh interesting, thank you for this!!

One thing I have thought for a while about this is that it makes sense to normalize by the canvas resolution, so that the mouse code runs similarly regardless of canvas size/position. I think this is most consistent with the ways other aspects of hydra work independently of pixel dimensions. I think this could be [0, 1], [-1, 1], or even [-0.5, 0.5] for the range of values when the mouse is over the canvas. I want to try it with some default functions to see what makes sense. In an ideal world, I think that a logical mapping is that `shape().scroll(() => mouse.x, () => mouse.y).out()` would cause the shape to travel directly under the mouse. However, I think the coordinates are flipped and this might just be confusing. 

Personally, I want to have some mouse code directly in hydra (same with things like rendering speed and fps, as well as the webrtc things and audio processing) as I like to think of these things as part of the synth / language even though they are not specifically part of the shader generation.  This makes it possible to write mouse and audio-reactive sketches that can be run in multiple places outside of the editor. Someone can always implement their own additional logic external to hydra-synth. I guess this is also dependent on getting non-global mode working again.

I was also thinking that touch points could be included in the mouse object, so that on mobile it would work using touch. Let me know what you think!

**Comment by ojack** [comment]
Okay, I think I better understand the mouse absolute position now! I also realized that I took this into a bigger conversation about the mouse in general.

I would like to approve this pull request, and move the broader mouse/pointer/touch discussion into a separate issue.

Is there anything else that needs to be done for this pull request to be ready? Let me know and I will merge + publish :)

**Comment by micuat** [comment]
as long as it's tested by someone it should be fine to be merged!

**Comment by micuat** [comment]
> In an ideal world, I think that a logical mapping is that `shape().scroll(() => mouse.x, () => mouse.y).out()` would cause the shape to travel directly under the mouse.

it would be in fact

```javascript
shape().scroll(() => 0.5-mouse.x/window.innerWidth, () => 0.5-mouse.y/window.innerHeight).out()
```

(with 0.5)

File: package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
     "test": "mocha",
     "dev": "budo ./dev/index.js --ssl --open --live --port 8000 --",
     "dev-insecure": "budo ./dev/index.js --open --live --port 8000 --",
-    "build-hydra": "browserify hydra-synth.js --standalone Hydra > dist/hydra-synth.js",
-    "build-shader-generator": "browserify shader-generator.js --standalone HydraShaders > dist/hydra-shaders.js",
+    "build-hydra": "browserify hydra-synth.js --standalone Hydra | tee dist/hydra-synth.js",
```
[the remark this answers is not in the document]
Comment (micuat) [reply package.json:10]: it was here https://github.com/ojack/hydra-synth/issues/80#issue-824479715

**Comment by ojack** [comment]
Uh oh, some reports of mouse y position not working anymore in Atom. Not sure if it is related.

**Pull request merged**